### PR TITLE
add clarification around null in state provider

### PR DESCRIPTION
**Pull request by jlf0dev**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

adds clarification to when `null` is emitted from state providers, and when there is no emit.
## Code changes

- **file.ext:** Description of what was changed and why

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: libs/common/src/platform/state/user-state.ts [hunk 1]
```diff
@@ -6,24 +6,25 @@ import { StateUpdateOptions } from "./state-update-options";
...
-   * Emits a stream of data.
-   */
-  readonly state$: Observable<T>;
+  /** Emits a stream of data. Emits null if the user does not have specified state. */
+  readonly state$: Observable<T | null>;
 
-  /**
-   * Emits a stream of data alongside the user id the data corresponds to.
-   */
+  /** Emits a stream of data alongside the user id the data corresponds to. */
```
Comment (justindbaur) [inline libs/common/src/platform/state/user-state.ts:14]:
```suggestion
  /** Emits a stream of data alongside the user id the data for that user. */
```

**Review by justindbaur — CHANGES_REQUESTED** [review]
This wording wasn't even from this PR I just read it again and noticed how bad it is. Other than that this is awesome thanks!

File: libs/common/src/platform/state/user-state.ts (hunk 1 above)
Comment (jlf0dev) [reply libs/common/src/platform/state/user-state.ts:14]: Think I'm still struggling to understand the wording of the suggested change 😁

Comment (justindbaur) [reply libs/common/src/platform/state/user-state.ts:14]: Yeah that is still maybe not so good.

```suggestion
  /** Emits a stream of tuples, with the first element being a user id and the second element being the data for that user. */
```

Do you think that helps?

Comment (jlf0dev) [reply libs/common/src/platform/state/user-state.ts]: Yep, that looks better. Committed!

**Review by MGibson1 — APPROVED** [review]

**Review by justindbaur — APPROVED** [review]

**Pull request merged**